### PR TITLE
[ticket/14066] Add template events to search_body.html

### PR DESCRIPTION
**Pull request by rxu**
<a href="https://tracker.phpbb.com/browse/PHPBB3-14066">PHPBB3-14066</a>.

### CODE REVIEW

**Comment by rxu** [comment]
!set Event

File: phpBB/styles/prosilver/template/search_body.html [hunk 1]
```diff
@@ -91,7 +101,9 @@ <h2 class="solo">{L_SEARCH}</h2>
 </div>
 
 </form>
+<!-- EVENT search_body_form_after -->
 
+<!-- EVENT search_body_recent_search_before -->
```
Comment (Nicofuma) [inline phpBB/styles/prosilver/template/search_body.html]: what is the use case of this event? (can't it be inside the IF?)

Comment (rxu) [inline phpBB/styles/prosilver/template/search_body.html]: In theory, to display some custom info unrelated to recent searches (or f.e. custom recent search loop).

File: phpBB/styles/subsilver2/template/search_body.html [hunk 2]
```diff
@@ -67,8 +80,9 @@
 
 	<br clear="all" />
 	<!-- ENDIF -->
+	<!-- EVENT search_body_recent_search_after -->
 
-	</div>
+</div>
```
Comment (marc1706) [inline phpBB/styles/subsilver2/template/search_body.html]: Unrelated change. Please don't change indentations.

File: phpBB/docs/events.md [hunk 3]
```diff
@@ -1165,6 +1165,14 @@ quickreply_editor_message_before
 * Since: 3.1.0-a4
 * Purpose: Add content before the quick reply textbox
 
+search_body_form_after
+===
+* Locations:
+    + styles/prosilver/template/search_body.html
+    + styles/subsilver2/template/search_body.html
+* Since: 3.1.6-RC1
```
Comment (marc1706) [inline phpBB/docs/events.md]: Could you update these to 3.1.7-RC1?

**Comment by rxu** [comment]
@marc1706 Updated #3805.

**Comment by rxu** [comment]
Rebased #3805.

**Comment by Nicofuma** [comment]
:+1:

File: phpBB/docs/events.md [hunk 4]
```diff
@@ -1314,6 +1322,102 @@ search_body_form_before
...
+
+search_body_search_query_prepend
+===
+* Locations:
+    + styles/prosilver/template/search_body.html
+    + styles/subsilver2/template/search_body.html
+* Since: 3.1.7-RC1
+* Purpose: Put content at the top of the search query fields set
+
+search_body_search_options_after
```
Comment (marc1706) [inline phpBB/docs/events.md]: search_body_search_options_... needs to be before search_body_search_query_...

**Comment by rxu** [comment]
@marc1706 Updated #3805.
